### DISCUSSION TITLE
Function () with no paramater and pipe, AND, function () with 1 paramater and `use`

### DISCUSSION THREAD

**Discussion by bobjoe12131** [gleam-lang/gleam#3573] [Questions & support] [answered]
```
fn foo(n) {
  n + 1
}

fn baz(fun) {
  foo(fun())
}

fn main() {
  let bar = 1
  let a = bar |> foo() // works
  let a = a |> foo // also works
  use <- baz() // also also works
  use <- baz // also also also works
  a
}
```
Why does the formatter and/or the compiler allow both parentheses and no parentheses?
Currently I use parentheses cause it looks like I am calling the function, and it does not get colored in vscode otherwise. but Ive seen others code use none.
Which should i use?

**Comment by lpil** [comment, marked as answer]
> Why does the formatter and/or the compiler allow both parentheses and no parentheses?

Because there's no way to know if it's safe to remove parens without performing type checking. It's recommended not to use parens.
